lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -702,20 +709,85 @@ def _get_run_job_params_from_existing_job(job: Job, user_id: str) -> Dict:
         # Then the next fields are job inputs top level requirements, app run parameters, and scheduler resource requirements
         return run_job_params
 
+    def _check_ws_perms(
+        self, runjob_params, new_batch_job, batch_params, as_admin=False
+    ):
+        """
+        Check a single job, a single batch job, or a retry_multiple request with a mix of different jobs.
+        """
+        if as_admin:
+            return self.sdkmr.check_as_admin(requested_perm=JobPermissions.WRITE)
+        # Batch params have the next highest
+        if new_batch_job:
+            if batch_params:
+                return self._check_workspace_permissions(batch_params.get("wsid"))
+            # No ws perms to check
+            return
+        # Single job runs
+        elif isinstance(runjob_params, dict):
+            return self._check_workspace_permissions(runjob_params.get("wsid"))
+        # Multiple independent job runs, think retry_multiple()
+        elif isinstance(runjob_params, list):
+            return self._check_workspace_permissions_list(
+                [job_param.get("wsid") for job_param in runjob_params]
+            )
+        else:
+            raise IncorrectParamsException(
+                "runjob_params must be a mapping or a list of mappings"
+            )
+
+    @staticmethod
+    def _propagate_wsid_for_new_batch_jobs(runjob_params, batch_params, new_batch_job):
+        """
+        For batch jobs, check to make sure the job params do not provide a wsid other than None or 0
+        Then Modify the run job params to use the batch params wsid, which may be set to None
+        """
+        if batch_params is None:
+            batch_params = {}
+
+        if new_batch_job:
+            batch_wsid = batch_params.get("wsid")
+            for i, runjob_param in enumerate(runjob_params):
+                runjob_param_wsid = runjob_param.get("wsid")
+                if runjob_param_wsid or runjob_param_wsid == 0:
+                    raise InvalidParameterForBatch()
+                # Do we do a deepcopy here in case the params point to the same obj?
+                runjob_params[i]["wsid"] = batch_wsid
+
+    def preflight(
+        self, runjob_params, batch_params=None, new_batch_job=False, as_admin=False
+    ):
+        if batch_params and not new_batch_job:
```

Comment:
Naively it seems to me that `batch_params` and `new_batch_job` are redundant - if `batch_params` is truthy you can just assume `new_batch_job` is True. I don't think I understand the purpose of the argument

Reply:
Never mind - `batch-params` can still be `None` or `{}` for a batch job. Could we document that though? I'm guessing I'm not the only person that'll get confused

Reply:
Added that to comments

Reply:
:+1: